// ordiscan/dist/schemas/brc20.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Brc20ActivitySchema = exports.Brc20BalanceSchema = exports.Brc20TokenSchema = exports.Brc20ActionSchema = exports.Brc20ActionTypeSchema = void 0;
const zod_1 = require("zod");
exports.Brc20ActionTypeSchema = zod_1.z.enum(["TRANSFER", "MINT", "DEPLOY"]);
exports.Brc20ActionSchema = zod_1.z.object({
    tick: zod_1.z.string(),
    type: exports.Brc20ActionTypeSchema,
});
exports.Brc20TokenSchema = zod_1.z
    .object({
    tick: zod_1.z.string(),
    minted: zod_1.z.number(),
    max_supply: zod_1.z.number(),
    price: zod_1.z.number().nullable(),
})
    .strict();
exports.Brc20BalanceSchema = zod_1.z.object({
    tick: zod_1.z.string(),
    balance: zod_1.z.number(),
});
exports.Brc20ActivitySchema = zod_1.z.object({
    ticker: zod_1.z.string(),
    type: exports.Brc20ActionTypeSchema,
    from_address: zod_1.z.string().nullable(),
    to_address: zod_1.z.string().nullable(),
    amount: zod_1.z.number(),
    inscription_id: zod_1.z.string(),
    timestamp: zod_1.z.string(),
});
