// ordiscan/dist/schemas/sat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RareSatBalanceSchema = exports.SatSchema = exports.SatributeSchema = void 0;
const zod_1 = require("zod");
exports.SatributeSchema = zod_1.z.enum([
    "UNCOMMON",
    "RARE",
    "EPIC",
    "LEGENDARY",
    "MYTHIC",
    "BLACK_UNCOMMON",
    "BLACK_RARE",
    "BLACK_EPIC",
    "BLACK_LEGENDARY",
    "BLACK_MYTHIC",
    "BLOCK_9",
    "BLOCK_78",
    "NAKAMOTO",
    "FIRST_TX",
    "VINTAGE",
    "PIZZA",
    "HITMAN",
    "PALINDROME",
    "ALPHA",
    "OMEGA",
]);
exports.SatSchema = zod_1.z
    .object({
    sat_number: zod_1.z.number(),
    satributes: zod_1.z.array(exports.SatributeSchema),
    creation_date: zod_1.z.string(),
    block_height: zod_1.z.number(),
    epoch: zod_1.z.number(),
    inscription_ids: zod_1.z.array(zod_1.z.string()),
})
    .strict();
exports.RareSatBalanceSchema = zod_1.z.object({
    satributes: zod_1.z.array(exports.SatributeSchema),
    ranges: zod_1.z.array(zod_1.z.array(zod_1.z.number())),
});
